### PR TITLE
Initial add of modularized arch guide content

### PR DESCRIPTION
**Pull request by maxwelldb**
* Fixing symlinks

* Renaming assemblies

* Updating assemblies in topic map

* Broad name, link, and organization changes.

* Beginning restructuring

### CODE REVIEW

**Comment by maxwelldb** [comment]
@kalexand-rh Here's another first pass. :1st_place_medal: 

We can look at the overlapping stuff and chunk/move/remove together, if you like.

**Comment by kalexand-rh** [comment]
@maxwelldb thanks! I'm going to merge this PR and open a follow-up later today. I think that there's some duplication to fix and procedural content to pull, but this seems like a good conversion of the source. I'll tag you in the follow-up so we can talk about some of the edits.

File: architecture/understanding-development.adoc [hunk 1]
```diff
@@ -0,0 +1,45 @@
+[id="understanding-openshift-development"]
+= Understanding {product-title} development
+include::modules/common-attributes.adoc[]
+:context: container-development
+toc::[]
+
+For many people, their first experience building and running containers is awesome.
```
Comment (mbearer) [inline architecture/understanding-development.adoc]: remove:

> For many people, their first experience building and running containers is awesome.

File: architecture/understanding-development.adoc [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+Often knowing little or nothing about containers, many people have been able to build a containerized application in just a few minutes, push it to a registry to make it available to anyone they chose, and run it from any Linux system with a container runtime. If they were just running individual applications on their local laptop, they might feel like containers give them everything they need.
```
Comment (mbearer) [inline architecture/understanding-development.adoc]: Remove:

> Often knowing little or nothing about containers, many people have been able to build a containerized application in just a few minutes, push it to a registry to make it available to anyone they chose, and run it from any Linux system with a container runtime. If they were just running individual applications on their local laptop, they might feel like containers give them everything they need.

File: architecture/understanding-development.adoc [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+But as thrilling as the first experience can be, building and running a single container manually just makes you want more. To make containers a viable entity for developing and running enterprise-quality applications, they needed to be surrounded by tools that allowed them to be:
```
Comment (mbearer) [inline architecture/understanding-development.adoc]: Rephrase this:

> But as thrilling as the first experience can be, building and running a single container manually just makes you want more. To make containers a viable entity for developing and running enterprise-quality applications, they needed to be surrounded by tools that allowed them to be:

to something like: 

> To fully leverage the capability of containers when developing and running enterprise-quality applications, ensure your environment is supported by tools that allow containers to be:

**Pull request merged**